src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -19,7 +44,21 @@ const useTriggerProcessing = (props: { serialNumber: string; trigger: triggerTyp
 
   useEffect((): any => {
     if (data?.applicationTriggerStates?.nodes[0]) {
-      const triggerRequested = data?.applicationTriggerStates?.nodes[0][trigger]
+      let triggerRequested
+      switch (triggerType) {
+        case 'applicationTrigger':
+          triggerRequested = data?.applicationTriggerStates?.nodes[0]?.applicationTrigger
```

Comment:
Probably need to check first if `data?.applicationTriggerStates?.nodes` is not empty

Reply:
Isn't that what this is doing at the top of the function?
```
if (data?.applicationTriggerStates?.nodes[0])
```

Reply:
No, I think it would crash there if there was no applicationTriggerStates

Reply:
Ah, okay looks like this will work:
`if (data?.applicationTriggerStates?.nodes?.[0])` 😎

Done a5cf6b6a

Reply:
Ah, I'm not sure this will solve this. Sorry! This is again checking for null, but you still try to access an index of the nodes array without checking if this array is empty! I've seen this happening before in many places with GraphQL queries returning empty arrays, is better to have a safe guard for this here. We will improve the error handling at some point, but we still want to make this easier for development, if something unexpected happens it would be easier to see the error instead of a hook that crashes which would happen in this case.

Reply:
> but you still try to access an index of the nodes array without checking if this array is empty!

That's what the `?.` before the index is for. It won't try and access the index if it doesn't exist!

Okay, I double-checked this and it seems fine: https://jsfiddle.net/CarlosNZ/z0e6jwg5/16/